Add base documentation (#1)
blackfireio/integration-heroku: README.md

The Blackfire Agent is a daemon processing the data collected by the Blackfire Probe, and sending it to the Blackfire servers. The Blackfire Probe is a language extension; it collects resources consumption metrics on profiled code.

The Blackfire CLI Tool provides a client with 2 main commands:

* An HTTP client wrapping cURL to [profile web based apps](https://blackfire.io/docs/cookbooks/profiling-http-via-cli);
* A client to [profile CLI Commands](https://blackfire.io/docs/cookbooks/profiling-cli).

## Requirements

- [Signup or login to Blackfire.io](https://blackfire.io/signup)
- Make sure the Blackfire Probe is installed (see installation procedures for each supported language below)
- [Make sure that the Blackfire servers can reach your servers](https://blackfire.io/docs/cookbooks/reverse-proxies).

### PHP

If your app is written in PHP, **you currently don't need to use this buildpack as it is already [integrated in the
official PHP buildpack](https://blackfire.io/docs/integrations/paas/heroku)**.

### Python

> Blackfire probe has to be installed via `pip`. As a result, you need to refer `blackfire` as dependency in your
> `requirements.txt` file, e.g.:

Reviewer: ```suggestion
Install the Blackfire Probe by referring `blackfire` as dependency in your `requirements.txt` file, e.g.:
```

# Heroku Buildpack: Blackfire

> **If your app is written in PHP, you currently don't need to use this buildpack** as it is already [integrated in the
> official PHP buildpack](https://blackfire.io/docs/integrations/paas/heroku).

This is the official Heroku Buildpack for [Blackfire](https://blackfire.io).

> Blackfire is the Code Performance Management solution for developers
> to find and fix performance bottlenecks in dev, test/staging and production.

Reviewer: ```suggestion
Blackfire Profiler is a Software-as-a-Service tool which measures how your code consumes resources at run-time. It enables to find performance bottlenecks and understand the code's behavior. It can be used in development, test/staging and production (with no overhead for end-users).

Intuitive visualizations enable you to browse through your call stack, and review Wall-time, CPU time, I/O time, Memory, Network calls, HTTP requests and SQL queries usage.
```

> This buildpack provides **Blackfire Agent** and **CLI Tool**. 

Reviewer: ```suggestion
This buildpack provides the **Blackfire Agent** and the **Blackfire CLI Tool**. Please refer to the Blackfire documentation for [more details on the Blackfire stack](https://blackfire.io/docs/reference-guide/faq#the-blackfire-stack).
```

> The Agent is the component that aggregates profile data collected by the *probe* from your application engine
> (PHP, Python, Go...), before sending it to Blackfire.io servers so that you can display and analyze it.

Reviewer: ```suggestion
The Blackfire Agent is a daemon processing the data collected by the Blackfire Probe, and sending it to the Blackfire servers. The Blackfire Probe is a language extension; it collects resources consumption metrics on profiled code.
```

> The CLI Tool provides a Blackfire client with 2 main commands:

Reviewer: ```suggestion
The Blackfire CLI Tool provides a client with 2 main commands:
```

* An HTTP client wrapping cURL to [profile web based apps](https://blackfire.io/docs/cookbooks/profiling-http-via-cli);
* A client to [profile CLI Commands](https://blackfire.io/docs/cookbooks/profiling-cli).

## Requirements

> First and foremost, **you need an account on https://blackfire.io to use this buildpack**.
> 
> You need to have **Blackfire probe** installed in your app.

Reviewer: ```suggestion
- [Signup or login to Blackfire.io](https://blackfire.io/signup)
- Make sure the Blackfire Probe is installed (see installation procedures for each supported language below)
- [Make sure that the Blackfire servers can reach your servers](https://blackfire.io/docs/cookbooks/reverse-proxies).
```

### PHP

If your app is written in PHP, **you currently don't need to use this buildpack as it is already [integrated in the
official PHP buildpack](https://blackfire.io/docs/integrations/paas/heroku)**.

### Python

> Blackfire probe has to be installed via `pip`. As a result, you need to refer `blackfire` as dependency in your

Reviewer: > Blackfire probe has to be installed via `pip`.
Does that give enough guidance? If I look at our docs, it seems further info would be welcome.

Instead of giving the install procedure here for the Agent, and in our docs for the Probe, shouldn't we have everything in our docs?

`requirements.txt` file, e.g.:

```
django
jinja2
gunicorn
blackfire
```

### Go

1. Import the Blackfire module in your code base:

   ```go
   import github.com/blackfireio/go-blackfire
   ```

2. [Update your code](https://blackfire.io/docs/integrations/go/sdk) to enable profiling in your Go app.

   Check that there are no errors in the logs. To debug problems, change the log level and the log file in the
   [probe configuration](https://blackfire.io/docs/configuration/go).

## Configuration

To set your config vars, you may use `heroku config:set` command:

```bash

> heroku config:set BLACKFIRE_SERVER_ID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

Reviewer: Looking at my previous comment, it would also enable to have the selection of actual credential values, instead of showing placeholders here?
